File: Programmers/2/62048. 멀쩡한 사각형/멀쩡한 사각형.cpp
#include <bits/stdc++.h>

using namespace std;

long long gcd(long long a, long long b)
{
    while (b != 0)
    {
        long long r = a % b;
        a = b;
        b = r;
    }
    
    return a;
}

long long solution(int w,int h) {
    long long W = w;
    long long H = h;
    long long answer = W * H;
    
    long long G = gcd(W, H);
    long long broken = W + H - G;
    
    return answer - broken;
}

File: Programmers/2/62048. 멀쩡한 사각형/멀쩡한 사각형.java
class Solution {
    public long solution(int w, int h) {
        long W = w;
        long H = h;
        long answer = W * H;
        
        long g = gcd(W, H);
        long broken = W + H - g;
        
        return answer - broken;
    }
    
    private long gcd(long a, long b) {
        while (b != 0) {
            long r = a % b;
            a = b;
            b = r;
        }
        return a;
    }    
}